WordPress/Classes/ViewRelated/Blog/Settings/SiteSettingsView.swift

```diff
@@ -0,0 +1,137 @@
+    var body: some View {
+        List {
+            sections
+        }
+        .listStyle(.insetGrouped)
+        .backport.refreshable {
+            await viewModel.refresh()
+        }
+        .navigationTitle(Strings.title)
+        .navigationBarTitleDisplayMode(.inline)
+        .toolbar {
+            ToolbarItem(placement: .navigationBarTrailing) {
+                if presentationMode.wrappedValue.isPresented {
+                    closeButton
+                }
+            }
+        }
+        .onReceive(viewModel.onDismissableError) {
+            SVProgressHUD.showDismissibleError(withStatus: $0)
+        }
+        .onAppear {
+            Task { await viewModel.refresh() }
+        }
+    }
+
+    @ViewBuilder
+    private var sections: some View {
+        Section(header: Text(Strings.Sections.general)) {
+            siteTitleRow
+        }
+    }
+
+    // MARK: - General
+
+    private var siteTitleRow: some View {
+        withAdminNavigationLink(destination: {
+            SettingsTextEditView(
+                value: settings.name,
+                placeholder: Strings.General.siteTitlePlaceholder,
+                onCommit: viewModel.updateSiteTitle
+            )
+            .navigationTitle(Strings.General.siteTitle)
+        }) {
+            SettingsCell(title: Strings.General.siteTitle, value: settings.name ?? Strings.General.siteTitlePlaceholder)
+        }
+    }
+
+    // MARK: - Helpers
+
+    @ViewBuilder
+    private func withAdminNavigationLink<T: View, U: View>(
+        @ViewBuilder destination: () -> T,
+        @ViewBuilder content: () -> U
+    ) -> some View {
+        if blog.isAdmin {
+            NavigationLink(destination: destination(), label: content)
+        } else {
+            content()
+        }
+    }
+
+    private var closeButton: some View {
+        Button(action: { presentationMode.wrappedValue.dismiss() }) {
+            Text(Strings.done)
+                .font(.body.weight(.medium))
+                .foregroundColor(Color.primary)
+        }
+    }
+}
+
+private struct SettingsCell: View {
+    let title: String
+    let value: String
+
+    var body: some View {
+        HStack {
+            Text(title)
+                .layoutPriority(1)
+            Spacer()
+            Text(value)
+                .foregroundColor(.secondary)
+        }
+        .lineLimit(1)
+    }
+}
+
+private struct SettingsTextEditView: UIViewControllerRepresentable {
+    let value: String?
+    let placeholder: String
+    var hint: String?
+    let onCommit: ((String)) -> Void
+
+    func makeUIViewController(context: Context) -> SettingsTextViewController {
+        let viewController = SettingsTextViewController(text: value ?? "", placeholder: placeholder, hint: hint ?? "")
+        viewController.onValueChanged = onCommit
+        return viewController
+    }
+
+    func updateUIViewController(_ uiViewController: SettingsTextViewController, context: Context) {
+        // Do nothing
+    }
+}
```

Comment:
I might be missing context but what is the purpose of all this? Is this the next screen in the flow which is a VC and we are rewriting the first screen in SwiftUI first?

Reply:
I'm prioritizing updating `SiteSettingsViewController` itself to make it easier to maintain it, but if I have time, I'll try to at least update the Objective-C screens that it invokes. `SettingsTextViewController` is relatively big, so it's a stopgap.

Reply:
@kean that makes sense. Thanks for explaining!